### PR TITLE
Migrate pallet-vesting benchmark to v2

### PR DESCRIPTION
**Pull request by AurevoirXavier**
Part of:

- #6202.

### CODE REVIEW

File: substrate/frame/vesting/src/benchmarking.rs [hunk 1]
```diff
@@ -70,322 +69,351 @@ fn add_vesting_schedules<T: Config>(
...
+		vest(RawOrigin::Signed(caller.clone()));
+
 		// Vesting schedule is removed!
-		assert_eq!(
-			Vesting::<T>::vesting_balance(&caller),
-			None,
-			"Vesting schedule was not removed",
-		);
-	}
+		assert_eq!(Pallet::<T>::vesting_balance(&caller), None, "Vesting schedule was not removed",);
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply substrate/frame/vesting/src/benchmarking.rs:131]: I don't think it use that version.

```
cargo +1.81 fmt

Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `comment_width = 100`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `reorder_impl_items = false`, unstable features are only available in nightly channel.
Warning: can't set `spaces_around_ranges = false`, unstable features are only available in nightly channel.
Warning: can't set `binop_separator = Back`, unstable features are only available in nightly channel.
Warning: can't set `match_arm_blocks = false`, unstable features are only available in nightly channel.
Warning: can't set `trailing_semicolon = false`, unstable features are only available in nightly channel.
Warning: can't set `trailing_comma = Vertical`, unstable features are only available in nightly channel.
```

[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply substrate/frame/vesting/src/benchmarking.rs:131]: There is no nightly-1.81 I think.

[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply substrate/frame/vesting/src/benchmarking.rs:131]: `cargo +nightly-2024-09-11 fmt` nothing changed.

[the remark this answers is not in the document]
Comment (rockbmb) [reply substrate/frame/vesting/src/benchmarking.rs:131]: Hm, I was being overzealous, then. Thanks for confirming!

File: substrate/primitives/panic-handler/src/lib.rs [hunk 2]
```diff
@@ -30,7 +30,7 @@ use std::{
 	cell::Cell,
 	io::{self, Write},
 	marker::PhantomData,
-	panic::{self, PanicInfo},
+	panic::{self, PanicHookInfo},
```
Comment (seadanda) [inline substrate/primitives/panic-handler/src/lib.rs:33]: Unrelated change - why is this in the PR?

Comment (AurevoirXavier) [reply substrate/primitives/panic-handler/src/lib.rs:33]: I'm ok to move this to a single PR

**Review by shawntabrizi — APPROVED** [review]

**Comment by AurevoirXavier** [comment]
> @AurevoirXavier Contributor did not properly post their account address.
> 
> Make sure the pull request description (or user bio) has: "{network} address: {address}".

Added.

File: substrate/primitives/panic-handler/src/lib.rs (hunk 2 above)
Comment (ggwpez) [reply substrate/primitives/panic-handler/src/lib.rs:33]: We will temporarily revert this, as it causes issues with the CI.
